test/unit/utils/general.spec.js

```diff
@@ -12,50 +12,4 @@ describe('General Utils', function() {
       expect(element).toBeFalsy();
     });
   });
-
-  describe('getTetherOptions', function() {
-    it('classes set correctly', function() {
-      const step = new Step({}, {
-        attachTo: { element: '.scroll-test', on: 'center' },
-        tetherOptions: {
-          classes: {
-            element: 'bar'
-          }
-        }
-      });
-
-      const tetherOptions = getTetherOptions(parseAttachTo(step), step);
-      expect(tetherOptions.classes.element).toBe('bar');
-    });
-
-    it('constraints can be overridden', function() {
```

Comment:
Should we replace these tests with some `popperOptions` tests? Would be good to make sure deeply nested options can be passed and override ours. I had issues with this in the past.

Reply:
I was a little back and forth on this, but should it be on the implementor to understand what they're passing in and use v2 docs?

Reply:
We should add tests for the options merging is what I mean. like `options.modifiers.foo.bar` and make sure that overrides our defaults and can still be passed down, and we don't clobber it. Does that make sense?

Reply:
I had a lot of issues before with using `Object.assign` and accidentally clobbering nested options.